### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by Bhagyesh22**

### CODE REVIEW

File: src/main/java/com/browserstack/pages/Bag.java [hunk 1]
```diff
@@ -4,19 +4,64 @@
...
 
-	public Bag(WebDriver driver) {
+
+	public Bag(WebDriver driver, WebDriverWait wait) {
 		this.driver = driver;
 		PageFactory.initElements(driver, this);
+		AddToCart(wait);
+	}
+
+	public void AddToCart(WebDriverWait wait) {
```
Comment (anirudha13) [inline src/main/java/com/browserstack/pages/Bag.java:29]: Method names must begin with a lower case letter.

File: src/test/java/com/browserstack/examples/config/WebDriverConfiguration.java [hunk 2]
```diff
@@ -31,9 +30,11 @@
     @JsonProperty("remoteDriver")
     private RemoteDriverConfig remoteDriverConfig;
 
+
     public List<Platform> getActivePlatforms() {
         List<Platform> activePlatforms = Collections.emptyList();
         switch (driverType) {
+            case dockerDriver:
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/config/WebDriverConfiguration.java:37]: Rename dockerDriver to onPremGridDriver.

File: src/test/java/com/browserstack/examples/config/WebDriverFactory.java [hunk 3]
```diff
@@ -48,7 +48,7 @@
     private static final String DEFAULT_BUILD_NAME = "browserstack-examples-junit4";
 
     private static WebDriverFactory instance;
-    private final WebDriverConfiguration webDriverConfiguration;
+    public final WebDriverConfiguration webDriverConfiguration;
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/config/WebDriverFactory.java:51]: Why is this public?

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java [hunk 4]
```diff
@@ -2,33 +2,35 @@
...
 public class WebDriverProviderRule extends TestWatcher {
-
     private static final Logger LOGGER = LoggerFactory.getLogger(WebDriverProviderRule.class);
-
     private static final String TEST_STATUS_SCRIPT = "browserstack_executor: {\"action\": \"setSessionStatus\", \"arguments\": {\"status\": \"%s\", \"reason\": \"%s\"}}";
-
     private String methodName;
     private WebDriver driver;
     private final WebDriverFactory webDriverFactory;
+    private WebDriverConfiguration webDriverConfiguration;
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:29]: Do not add this. Add the required methods to the WebDriverFactory itself.

File: src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java [hunk 5]
```diff
@@ -20,12 +24,19 @@
...
     @Parameterized.Parameters
-    public static Iterable<Object[]> data() {
-        WebDriverFactory webDriverFactory = WebDriverFactory.getInstance();
+    public static Iterable<Object[]> data() throws IOException {
+        String capabilitiesConfigFile = System.getProperty(CAPABILITIES_FILE_PROP, DEFAULT_CAPABILITIES_FILE);
+        LOGGER.debug("Using capabilities configuration from FILE :: {}", capabilitiesConfigFile);
+        URL resourceURL = AbstractWebDriverTest.class.getClassLoader().getResource(capabilitiesConfigFile);
+
+        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+        webDriverConfiguration = objectMapper.readValue(resourceURL, WebDriverConfiguration.class);
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java:37]: Why do we need to parse the capability file here again?

**Review by anirudha13 — CHANGES_REQUESTED** [review]

File: src/main/java/com/browserstack/pages/Bag.java (hunk 1 above)
Comment (Bhagyesh22) [reply src/main/java/com/browserstack/pages/Bag.java]: Done, renamed all methods that were starting with uppercase letter

File: src/test/java/com/browserstack/examples/config/WebDriverConfiguration.java (hunk 2 above)
Comment (Bhagyesh22) [reply src/test/java/com/browserstack/examples/config/WebDriverConfiguration.java]: Renaming Done

File: src/test/java/com/browserstack/examples/config/WebDriverFactory.java (hunk 3 above)
Comment (Bhagyesh22) [reply src/test/java/com/browserstack/examples/config/WebDriverFactory.java]: I was using one of the method of WebDriverConfig class and hence to initialise this object in other class , i made it public, but now i have made it private and added that method in webdriverfactory class

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java (hunk 4 above)
Comment (Bhagyesh22) [reply src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java]: Done , i have removed   private WebDriverConfiguration webDriverConfiguration and added method in WebDriverFactory class

File: src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java (hunk 5 above)
Comment (Bhagyesh22) [reply src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java]: I have removed that parsing logic from there

File: src/main/java/com/browserstack/pages/FavUserHomePage.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class FavUserHomePage {
+
+    WebDriver driver;
+    public final static String OrderButtonLocator = "orders";
```
Comment (anirudha13) [inline src/main/java/com/browserstack/pages/FavUserHomePage.java:14]: CONSTANT

File: src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java [hunk 7]
```diff
@@ -1,16 +1,21 @@
...
 
+import com.browserstack.examples.config.WebDriverFactory;
 import org.junit.Rule;
 import org.junit.runners.Parameterized;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.browserstack.examples.config.Platform;
-import com.browserstack.examples.config.WebDriverFactory;
+import com.browserstack.examples.config.WebDriverConfiguration;
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java:15]: Unused imports.

File: src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java [hunk 8]
```diff
@@ -1,16 +1,21 @@
...
 import com.browserstack.examples.rule.WebDriverProviderRule;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java:18]: Unused imports.

**Review by anirudha13 — CHANGES_REQUESTED** [review]

File: src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java (hunk 8 above)
Comment (Bhagyesh22) [reply src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java]: Removed all the unused imports

File: src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java (hunk 7 above)
Comment (Bhagyesh22) [reply src/test/java/com/browserstack/examples/tests/AbstractWebDriverTest.java]: Removed them

File: src/main/java/com/browserstack/pages/FavUserHomePage.java (hunk 6 above)
Comment (Bhagyesh22) [reply src/main/java/com/browserstack/pages/FavUserHomePage.java]: All variables that are declared as final renamed to be in ALL CAPS

File: src/main/java/com/browserstack/pages/Bag.java [hunk 9]
```diff
@@ -4,23 +4,64 @@
...
 public class Bag {
 
 	WebDriver driver;
 
 	public final static String checkoutselector = "#__next > div > div > div.float-cart.float-cart--open > div.float-cart__content > div.float-cart__footer > div.buy-btn";
+	public final static String SignInSelector = "signin";
+	public final static String iPhone12Locator = "#\\31  > div.shelf-item__buy-btn";
+	public final static String iPhone12MiniLocator = "#\\32  > div.shelf-item__buy-btn";
+	public final static String iPhone12ProMaxLocator = "#\\33  > div.shelf-item__buy-btn";
+	public final static String OrderButtonLocator = "orders";
```
[the remark this answers is not in the document]
Comment (Bhagyesh22) [reply src/main/java/com/browserstack/pages/Bag.java]: All variables that are declared as final renamed to be in ALL CAPS

**Pull request merged**